### ISSUE TITLE
Allow multiple HTTP proxies on different ports

### ISSUE THREAD

**Issue by ravenclaw900** [fatedier/frp#2501]
**The solution you want**

The ability to have multiple HTTP proxies on different remote ports.

**Alternatives considered**

Giving the HTTPS type proxy the same options as HTTP type proxies, but that would still only allow 2 remote ports.

**How to implement this function**

Remove `vhost_http_port` and replace it with `remote_port` on the client, like with TCP proxies.

**Application scenarios of this function**

Anything requiring multiple HTTP proxies on different ports, (e.g. a Synapse server)

**Comment by bingtianbaihua** [comment]
I don't know this scenarios, maybe it's not a common feature.

**Comment by ravenclaw900** [comment]
bump

**Comment by green-codes** [comment]
I'm running many http proxies on the same host with no problem. Just use type=http(s) and different ports for each service.

**Comment by ravenclaw900** [comment]
bump

**Comment by ravenclaw900** [comment]
bump

**Comment by ravenclaw900** [comment]
bump

**Comment by ravenclaw900** [comment]
bump

**Comment by ravenclaw900** [comment]
@green-codes, yes, but can't you only have one vhost? Doesn't that mean that all of the proxies would go to the same port?

**Comment by fatedier** [comment]
It's not necessary to always set type `http` for your service. Just use `tcp` instead.

**Comment by ravenclaw900** [comment]
Yes, however you do need type `http` to set headers, right? My situation is that I have 2 `http` proxies that need to have different headers set on them.

**Comment by fatedier** [comment]
We want to support this scenario in frp v2 since it's very different with current architecture.
